File: java/src/main/java/mx/iteso/ut/Queso.java
package mx.iteso.ut;
/** Interfaz del queso.*/
public interface Queso {
    /**
     * @return esta derretido?.
     */
     boolean isMelted();

    /**
     *
     * @return temperatura actual.
     */
     int getCurrentTemperature();

    /**
     *
     * @return temperatura a la que se derrite queso.
     */
     int getMeltingTemperature();

    /**
     *
     * @param temp temperatura de la quesadilla.
     */
     void setCurrentTemperature(int temp);

    /**
     *
     * @param melted si el queso esta derretido.
     */
     void melt(boolean melted);
}

File: python/quesadilla/main/Queso.py
class Queso:

    # Constructor de Python
    def __init__(self, melted=None, temperature=None, melting_temperature=20):
        self.melted = melted
        self.temperature = temperature
        self.melting_temperature = melting_temperature

    def is_melted(self):
        return self.melted

    def get_current_temperature(self):
        return self.temperature

    def get_melting_temperature(self):
        return self.melting_temperature

    def set_current_temperature(self, arg):
        self.temperature = arg

    def melt(self, arg):
        self.melted = arg
